packages/server/src/orders/database/OrderSchema.ts

```diff
@@ -0,0 +1,74 @@
+import { Prop, raw, Schema, SchemaFactory } from '@nestjs/mongoose';
+import mongoose, { Document } from 'mongoose';
+
+import { Address } from '../../addresses/database/AddressSchema';
+import { User } from '../../users/database/UserSchema';
+
+export type OrderDocument = Order & Document;
+
+const ORDER_CONSTANTS = Object.freeze({
+  COMMENT: Object.freeze({
+    MAX_LENGTH: 300,
+  }),
+  PRICE: Object.freeze({
+    MIN: 0,
+  }),
+  COUNT: Object.freeze({
+    MIN: 1,
+  }),
+});
+
+class SubOrder {
+  @Prop({ required: true })
+  deliveryDate: Date;
+
+  @Prop({ required: true })
+  hourStart: Date;
+
+  @Prop({ required: true })
+  hourEnd: Date;
+
+  @Prop({ default: false })
+  delivered: boolean;
+}
+
+@Schema({
+  collection: 'order',
+})
+class Order {
+  readonly id: string;
+
+  @Prop({ type: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Address' }], required: true })
```

Comment:
Tutaj podobnie jak w review User Schema:
[https://github.com/CodersCamp2021-HK/CodersCamp2021.Project.Fullstack/pull/36#discussion_r816018269](Komentarz Tomka i Marty)
Zamiast importować całe mongoose itp to powinno wystarczyć:
```suggestion
  @Prop({ type: [{ type: ObjectId, ref: 'Address' }], required: true })
```

Reply:
rzeczywiście widziałam waszą konwersacje. Słuszna uwaga!